Aquarius Plus console library updates

iratahack opened the pull request:
Changes included in this PR are:

1. On the aqplus, the border fill character and color can be remapped to the end of char and color RAM. This allows the first line of the console to be used. This remap is now enabled by default in crt0. So when building for the aqplus, there are 25 console lines not 24 as with the standard aquarius.
2. To correctly support 80 column mode, some of the stdio routines needed to be updated to account for the additional char and color memory. Basically, cls, scroll, xypos, and printc have been updated to detect the current mode, 40 or 80 column and call the appropriate routine. 
3. Updated aqplus.h to include the hardware register bit definitions.
4. New example program under example/aquarius to show some interrupt and 40/80 column console I/O. See attached images.

![screenshot2](https://github.com/user-attachments/assets/1cf478e1-307a-4ecc-840e-0c4d42af17b4)
![screenshot1](https://github.com/user-attachments/assets/e5e27059-a708-4501-9565-2d6afd0919b5)

suborb commented:
Can you also update target/aquarius/stdio/generic_console_ioctl to:

* Set the number of rows correctly
* Set the number of columns for 80 column mode

iratahack commented:
> Can you also update target/aquarius/stdio/generic_console_ioctl to:
> 
> * Set the number of rows correctly
> * Set the number of columns for 80 column mode

I've updated generic_console_ioctl and the isr example to use the get IOCTL to display the console size. I also updated the bordercolor routine to handle 40 and 80 column mode as the border color char  is in a different location.

iratahack commented on examples/aquarius/isr/main.c:
```diff
@@ -0,0 +1,111 @@
+{
+    unsigned int val;
+
+    console_ioctl(IOCTL_GENCON_CONSOLE_SIZE, &val);
```
Get the console size and display it.

iratahack commented on lib/target/aquarius/classic/aquarius_crt0.asm:
```diff
@@ -76,7 +85,15 @@ l_dcal:
+IF CLIB_AQUARIUS_PLUS = 1
+    ; Remap the border color character for aqplus
+    in      a, (IO_VCTRL)
+    or      VCTRL_REMAP_BC
```
Should 40 column mode be forced here so we have a known starting point for the video hardware?

iratahack commented:
Any other 40/80 column console changes I should make?

iratahack commented on libsrc/target/aquarius/stdio/generic_console_ioctl.asm:
```diff
@@ -1,126 +1,148 @@
-    ld      a,CLIB_GENCON_CAPS_TEXT
-    ld      (generic_console_caps),a
-    call    set_default_palette
+        ; Remap the border color character for aqplus
```
@suborb . It's not showing here for some reason, but I added the code to code_crt0_init section so the hardware and the library should get sync'd during init.

```
        SECTION code_crt_init

        EXTERN  set_default_palette

        ; On an Aquarius+ we modify the caps so that we can define the font
        ld      c, CLIB_AQUARIUS_PLUS
        rr      c
        jr      nc, not_plus

        call    set_default_palette

        ; Remap the border color character for aqplus
        in      a, (IO_VCTRL)
        or      VCTRL_REMAP_BC
        ld      e, a
        ld      a, IOCTL_GENCON_SET_MODE
        ; This will setup the mode and screen size
        ; so that it matches the hardware during init.
        call    ck_mode
```

suborb replied:
The `code_crt_init` section is only included if any of the `PUBLIC` definitions in the file are required.

What this means is that this initialisation is only included if the user calls `console_ioctl()`  (probably an existing issue looking at the palette issue)

iratahack replied:
Ohhh, got it. I was thinking it would be pulled in if --generic-console was used on the command line.

Any thoughts on how to handle this? I'll also take a look at some of the other targets to see how they handle this.

suborb replied:
I’ve tended to put them into generic_console.asm to handle that case. 

You’ll need to move the CAPS defc out  as
